app.model/app.model.js

```diff
@@ -60,15 +60,23 @@ exports.addVoteToArticle = (inc_votes, articleId) => {
   if (!inc_votes || typeof inc_votes !== "number" || isNaN(inc_votes)) {
     return Promise.reject({ status: 400, msg: "Bad Request" });
   }
-  return dataBaseConnection.query("SELECT * FROM articles").then((articles) => {
-    const allArticles = articles.rows;
-    const articleToPatch = allArticles.filter((article) => {
-      return article.article_id === parseInt(articleId);
-    });
-    if (articleToPatch.length === 0) {
-      return Promise.reject({ status: 404, msg: "Not Found" });
-    }
-    articleToPatch[0].votes += inc_votes;
-    return articleToPatch;
-  });
+    return dataBaseConnection
+      .query("SELECT * FROM articles WHERE article_id = $1", [articleId])
+      .then((result) => {
+        const articleToPatch = result.rows[0]; 
+        if (!articleToPatch) {
+          return Promise.reject({ status: 404, msg: "Not Found" });
```

Comment:
With our custom errors like this, we can afford to be more specific because we know exactly what causes them. The more info the better, and a descriptive message can go a long way in terms of being helpful on a frontend